### DISCUSSION TITLE
Odds setting

### DISCUSSION THREAD

**Discussion by vayner25** [MrABCDevelopment/RandomLootChest#7] [Q&A]
I have a problem with item spawning. I have prescribed the items and put them in the chances of dropping all the items under the highest chance of failing. The very first ones in the lists from a high chance drop out. 
9 Item Drop Examples:
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/2572d671-b056-4ab3-b4c2-e4adc18c7733)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/e14a5c6f-75d7-4a68-b92d-371066ae327f)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/cd06ce1d-a4fa-4d3d-8e40-eddce14c76b5)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/2a665de5-b96e-4589-8a64-6adb970bcacc)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/e32811ea-be4a-42ff-9e16-200541dc47fd)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/08cba400-ef89-45bf-a1ec-6eee1b68146b)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/6717779a-d8da-4f34-9f1b-d5a84fd4f9d3)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/290dc326-d2ac-4268-82bc-8dd32a8032ba)
My file settings chance: 
[default.zip](https://github.com/MrABCDevelopment/RandomLootChest/files/11959103/default.zip)
There are plenty of other items on this list that should drop, but they don't. I indicated less chances, since only the first ones fell out on high.

**Comment by MrABCDevelopment** [comment]
Currently, this system of chances, randomizing items, etc., loops all the items in the list and one by one checks if the item's chance is higher or equal, if it's higher or equal, then you'll see in the chest inventory that item. If there are more than x items from the list (you can set this using MaxItems option), other items from the list will be passed without adding them to the inventory.

**Reply by vayner25** [reply]
Help me to correctly compose the percentages of chances so that almost all items of one chance drop out, as well as the other. Let's say there will be haly, common, rare, exotic, legendary

**Comment by vayner25** [comment]
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/06dfa3f5-e275-459f-81dc-ff045ef43153)
the chances are small, as in the file, but there are many

**Reply by vayner25** [reply]
this my inventory

**Comment by MrABCDevelopment** [comment]
I suggest to put all items with small chance to drop on the top of the configuration section in chest file, so they have more chance to appear.

**Reply by vayner25** [reply]
Fine.  I will put the less rare items up and the rarest down, but what odds do I need to put down if I have 5 different types of rarity

**Reply by MrABCDevelopment** [reply]
You should put the rarest items up and less rare items down and finally check the results.

**Reply by vayner25** [reply]
Ok

**Reply by vayner25** [reply]
I'll try and post the result

**Comment by vayner25** [comment]
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/46d4f6bc-f011-4b0f-84ed-06f426865828)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/8e920dc8-39c5-4822-b121-ea3e2ff3e7db)
ERROR

**Reply by MrABCDevelopment** [reply]
Because this configuration of item is missing 2 attributes that are needed to work properly: **Extended** and **Upgraded**.

**Reply by vayner25** [reply]
i added them but nothing changed
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/bc9eff4e-8df5-48d5-9642-c8c9d39855b6)

**Reply by MrABCDevelopment** [reply]
My bad, you should use **PotionEffect** attribute, not PotionType

**Comment by vayner25** [comment]
If set to "MaxItems = 32" loot of different rarity drops out, if set to "MaxItems = 6" loot falls out of the first two rarities. It is impossible to see the 3rd level of rarity.
[default.zip](https://github.com/MrABCDevelopment/RandomLootChest/files/11995306/default.zip)

**Reply by MrABCDevelopment** [reply]
As I mentioned earlier - you should put the rarest items on the top of the contents list with the lowest chances to drop and items that are more likely to drop, they should be at the bottom. For example:

```
Contents:
  brown_mushroom:
    Material: brown_mushroom
    Amount: 1
    Chance: 0.01
  red_mushroom:
    Material: red_mushroom
    Amount: 1
    Chance: 0.05
  apple:
    Material: apple
    Amount: 1
    Chance: 0.1
  apple2:
    Material: apple
    Amount: 2
    Chance: 0.2
  wooden_sword:
    Material: wooden_sword
    Amount: 1
    Chance: 0.3
  stone_sword:
    Material: stone_sword
    Amount: 1
    Chance: 0.3
```

**Reply by vayner25** [reply]
I'm going to try now

**Reply by vayner25** [reply]
[default.zip](https://github.com/MrABCDevelopment/RandomLootChest/files/11995501/default.zip)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/dcf5cf7a-dc73-47f1-85f7-53e0e320a884)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/57262261-9183-4e3e-a31b-8728f1f43f8f)
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/89d061c4-84d3-4d1a-91a0-a58087be1b8f)

I did as you suggested and it just doesn't want to work. The most rare fall out most of all, there are almost no ordinary ones. An example of setting up a file threw off and attached screenshots of the loot for this file

**Reply by vayner25** [reply]
But if I change "MaxItems" to 32 in the file, then in general the drop chances are correct.

**Reply by MrABCDevelopment** [reply]
MaxItems option doesn't affect chances that are randomized and checked all the time. If you have these 6 items in the chest and option MaxItems is set to 6, then this chest won't fill with other items. Currently, this plugin is used in several servers by my friend and everything works well to this day.

**Reply by vayner25** [reply]
I use more items in the file, but I limited the number of spawns to 6. I noticed a pattern, if you set maxitems to 6, then only the very first items appear and very rarely, which are at the end of the list, if you raise the number, the spawn of items becomes larger and appears  even those at the end of the file.  You can take my file and test it on your server, or contact me and I'll show you a screen demo.

**Reply by vayner25** [reply]
I managed to set up

**Comment by vayner25** [comment]
This Error
![image](https://github.com/MrABCDevelopment/RandomLootChest/assets/69541777/326cbbc3-8cf7-4d38-a3d5-5f491ee5c518)
How to create enchantment books?

**Reply by MrABCDevelopment** [reply]
Currently, there is no option to create enchantment books, but this will change in the next big update.

**Reply by vayner25** [reply]
Thx. Will wait

**Comment by MrABCDevelopment** [comment]
There are few updates, I recommend install the latest version and check on your own all those things that are added to this plugin.
